docs/user/config-file/v2.rst

```diff
@@ -418,6 +418,42 @@ We don't currently support PPA's or other custom repositories.
    When possible avoid installing Python packages using apt (``python3-numpy`` for example),
    :ref:`use pip or Conda instead <guides/reproducible-builds:pinning dependencies>`.
 
+
+build.jobs
+``````````
+
+Commands to be run before or after a Read the Docs pre-defined job.
+This allows you to run custom commands at a particular moment in the build process.
```

Comment:
```suggestion
Commands to be run before or after a Read the Docs pre-defined build step.
This allows you to run custom commands at various times in the build process.

A few common use cases for this area:

* Modifying the state of the checkout of your repository
* Stopping a build by returning a non-zero exit code on certain conditions 
* Running a command to process generate markup files prior to building
```

Thinking more about this, we probably need to define this feature a lot more explicitly in the docs somewhere, particularly around exit codes.

Reply:
I didn't mention use cases here because none of the other config keys do that. Instead, I suggested writing a guide explaining what are the known use cases we have already and expanding on each of them clearly.

I see this page as reference documentation that only explains what are the possible/valid config keys and values of each of them, but it does not talk about use cases. I'd include the link to that guide page here once we write it, tho, as we do with `build.tools.python` and the Conda Support link (https://docs.readthedocs.io/en/stable/config-file/v2.html#build-tools-python)

What do you think?